tests/parse_self.rs

```diff
@@ -50,3 +53,46 @@ fn test_parse_self_debug_info() {
         }
     }
 }
+
+#[test]
+fn test_parse_self_debug_line() {
+    let debug_info = read_section("debug_info");
+    let debug_info = DebugInfo::<LittleEndian>::new(&debug_info);
+
+    let debug_abbrev = read_section("debug_abbrev");
+    let debug_abbrev = DebugAbbrev::<LittleEndian>::new(&debug_abbrev);
+
+    let debug_line = read_section("debug_line");
+    let debug_line = DebugLine::<LittleEndian>::new(&debug_line);
+
+    for unit in debug_info.units() {
+        let unit = unit.expect("Should parse the unit OK");
+
+        let abbrevs = unit.abbreviations(debug_abbrev)
+            .expect("Should parse abbreviations");
+
+        let mut cursor = unit.entries(&abbrevs);
+
+        let unit_entry = cursor.current_ref()
+            .expect("Should have a root entry")
+            .expect("and it should parse OK");
+
+        if let Some(attr) = unit_entry.attrs()
+            .map(Result::unwrap)
+            .find(|at| at.name() == DW_AT_stmt_list) {
+            // For whatever reason, rustc generated DW_FORM_data4 typed
+            // attributes for DW_AT_stmt_list attributes, when it seems to me
+            // like the more correct choice would be DW_FORM_sec_offset (the
+            // spec seems to agree -- see pages 214-215). Because of this, we
+            // have to turn the data into an offset.
```

Comment:
I think rustc defaults to DWARF 2.0, which did not have DW_FORM_sec_offset.  We probably should include test fixtures for both versions. This is also why there is no .debug_types section.